spec/models/form/lettings/questions/period_spec.rb

```diff
@@ -8,4 +8,39 @@
   it "has the correct hint" do
     expect(question.hint_text).to eq("Select how often the household is charged. This may be different to how often they pay.")
   end
+
+  it "has correct answer options" do
+    expect(question.answer_options).to eq(
+      {
+        "2" => { "value" => "Every 2 weeks" },
+        "3" => { "value" => "Every 4 weeks" },
+        "4" => { "value" => "Every calendar month" },
+        "9" => { "value" => "Weekly for 46 weeks" },
+        "8" => { "value" => "Weekly for 47 weeks" },
+        "7" => { "value" => "Weekly for 48 weeks" },
+        "6" => { "value" => "Weekly for 49 weeks" },
+        "5" => { "value" => "Weekly for 50 weeks" },
+        "11" => { "value" => "Weekly for 51 weeks" },
+        "1" => { "value" => "Weekly for 52 weeks" },
+        "10" => { "value" => "Weekly for 53 weeks" },
+      },
+    )
+  end
+
+  context "when managing organisation has rent periods" do
+    let(:managing_organisation) { create(:organisation, skip_rent_period_creation: true) }
+    let(:log) { create(:lettings_log, managing_organisation:) }
+
+    before do
+      create(:organisation_rent_period, organisation: managing_organisation, rent_period: 3)
+    end
```

Comment:
Did you try just building the log?
I think a normal build, would build an associated user, and the user factory would build an organisation and the log would be assigned to the user and the user's org and the org would have all rent periods built and added to its organisation_rent_periods attribute
```suggestion
    let(:log) { build(:lettings_log) }

    before do
      log.managing_organisation.organisation_rent_periods.filter! { |period| period.rent_period == 3 }
    end
```
I feel like this would work, but I might easily be wrong

Reply:
I think it might need to be created (?) cause this doesn't return any rent periods for the managing org at all

Reply:
oh yeah I looked at the rent_periods method and it actually uses pluck, which will force it to do database things